### PR TITLE
Expanding SaaS Content

### PR DESCRIPTION
**Pull request by erikmarr**
## Purpose of this pull request

This pull request (PR) updates the existing SaaS content based on feedback from writers and contributions from the development team.

## Magento release version

- [x] 2.4.x

   Specify a patch release number, if applicable:

- [ ] 2.3.x

   Specify a patch release number, if applicable:

## Product editions

Is this update specific to a product edition?

- [ ] Magento Open Source only
- [ ] Magento Commerce only

Is this update specific to an installed feature extension

- [ ] B2B extension
- [ ] Other feature set, please specify:

## Affected documentation pages

- https://docs.magento.com/user-guide/configuration/services/saas.html
- https://docs.magento.com/user-guide/system/saas.html

whatsnew
Expanded documentation about connecting to [SaaS services in Magento](https://docs.magento.com/user-guide/system/saas.html).

### CODE REVIEW

**Review by ccondra — APPROVED** [review]
This looks good!  One thing you might do is include the new topic in the PR Desc and also add a "whatsnew"

**Comment by erikmarr** [comment]
Thanks @ccondra! Updated PR.

File: src/system/saas.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+title: Magento Services
+---
+
+Some Magento features are deployed as SaaS services. To use these services, you must connect your Magento instance to these services using an API key and specifying the SaaS environment in the Magento configuration.
+
+## Magento API key {#apikey}
+
+When you create a Magento account, identified by a MageID, you can generate a Magento API key. To use Magento Commerce SaaS services, such as Product Recommendations, the license-holder must generate the API key in order to pass entitlement validation. This API key can then be passed to the systems integrator or development team that manages the SaaS projects and environments on behalf of the license-holder. If you are a Magento solution partner, the contract signer should generate the API key.
```
Comment (misha-kotov) [inline src/system/saas.md:9]:
```suggestion
When you create a Magento account, identified by a MageID, you can generate a Magento API key. To use Magento Commerce SaaS services, such as Product Recommendations, the license-holder must generate the API key in order to pass entitlement validation. This API key can then be passed to the systems integrator or development team that manages the SaaS projects and environments on behalf of the license-holder. If you are a solution integrator, the signer of the Magento partner contract should generate the API key.
```

File: src/system/saas.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+1. Generate and copy the API key for **Production**.
+
+   ![]({% link images/images/get-api-key.png %}){: .zoom}
+   _Get API Key_
+
+1. With the API key, you can now [create a SaaS Environment](#createsaasenv).
+
+## SaaS projects and environments {#saasenv}
+
+Magento SaaS services are made up of SaaS projects and environments. SaaS projects and environments are separate but analogous concepts to Magento Cloud projects and environments. A SaaS project is a grouping of SaaS environments. A SaaS environment defines a data space that is unique to your Magento instance. The SaaS environment is used to collect and store data that enables Magento services to work. Some of this data may be exported from Magento and some may be collected from shopper behavior on the storefront. That data is then stored in a secure cloud location and not on the Magento server.
```
Comment (misha-kotov) [inline src/system/saas.md:26]:
```suggestion
Magento SaaS services are made up of SaaS projects and environments. SaaS projects and environments are separate but analogous concepts to Magento Cloud projects and environments. A SaaS project is a grouping of SaaS environments. A SaaS environment defines a data space that is used to collect and store data that enables Magento services to work. Some of this data may be exported from Magento and some may be collected from shopper behavior on the storefront. That data is then persisted to secure cloud storage and not to the Magento servers.
```

**Review by misha-kotov — APPROVED** [review]

**Comment by erikmarr** [comment]
running tests

**Pull request merged**